src/Application/Economy/Commands/BalanceTransactionCommand.cs

```diff
@@ -8,13 +9,16 @@
 namespace Application.Economy.Commands;
 
 
-public sealed record BalanceTransactionCommand(User sender, User reciver, decimal amount) : IRequest<bool>;
+public sealed record BalanceTransactionCommand(UserId sender, UserId reciver, decimal amount) : IRequest<bool>;
 
 public sealed class TransactionValidator : AbstractValidator<BalanceTransactionCommand>
 {
     public TransactionValidator()
     {
-        RuleFor(x => x.amount).LessThanOrEqualTo(0).WithMessage("amount can not be 0 or less");
+        RuleSet("async", () => { RuleFor(x => x.amount)
```

Comment:
u dont need async validators for simple synchronous rules, however you will need the async ruleset to take place when you validate that the users are indeed found in the database. check the comment